jbpm-services/jbpm-services-ejb/jbpm-services-ejb-timer/src/main/java/org/jbpm/services/ejb/timer/EJBTimerScheduler.java

```diff
@@ -97,112 +94,78 @@ public void executeTimerJob(Timer timer) {
-        } catch (Exception e1) {
-            logger.warn("There was a problem during timer removal {}", ejbTimerJob.getTimerJobInstance(), e1);
-        }
-    }
-
-    private void recoverTimerJobInstance(EjbTimerJob ejbTimerJob, Timer timer, Exception e) {
-        if (isSessionNotFound(e)) {
-            logger.warn("Trying to recover timer. Not possible due to process instance is not found. More likely already completed. Timer {} won't be recovered", ejbTimerJob.getTimerJobInstance(), e);
+    private void recoverTimerJobInstance(EjbTimerJob ejbTimerJob, Timer timer,  Exception cause) {
+        Transaction<TimerJobInstance> tx;
+        if (isSessionNotFound(cause)) {
             // if session is not found means the process has already finished. In this case we just need to remove
             // the timer and avoid any recovery as it should not trigger any more timers.
-            removeUnrecoverableTimer(ejbTimerJob, timer);
-            return;
+            tx = timerJobInstance -> {
+                logger.warn("Trying to recover timer. Not possible due to process instance is not found. More likely already completed. Timer {} won't be recovered", timerJobInstance, cause);
+                if (!removeJob(timerJobInstance.getJobHandle(), timer)) {
+                    logger.warn("Session not found for timer {}. Timer could not removed.", timerJobInstance);
+                }
+            };
         }
-
-        if (ejbTimerJob.getTimerJobInstance().getTrigger().hasNextFireTime() != null) {
+        else if (ejbTimerJob.getTimerJobInstance().getTrigger().hasNextFireTime() != null) {
             // this is an interval trigger. Problem here is that the timer scheduled by DefaultTimerJobInstance is lost
             // because of the transaction, so we need to do this here.
-            try {            
-               
-                logger.warn("Execution of time failed Interval Trigger failed. Skipping {}", ejbTimerJob.getTimerJobInstance());
-                Transaction<TimerJobInstance> tx = timerJobInstance -> {
-                    if (this.removeJob(timerJobInstance.getJobHandle(), null)) {
-                        this.internalSchedule(timerJobInstance);
-                    } else {
-                        logger.debug("Interval trigger {} was removed before rescheduling", ejbTimerJob.getTimerJobInstance());
-                    }
-                };
-                transaction(tx, ejbTimerJob.getTimerJobInstance());
-            } catch (Exception e1) {
-                logger.warn("Could not schedule the interval trigger {}", ejbTimerJob.getTimerJobInstance(), e1);
-            }
-            return;
+            tx = timerJobInstance -> {
+                logger.warn("Execution of time failed Interval Trigger failed. Skipping {}", timerJobInstance);
+                if (removeJob(timerJobInstance.getJobHandle(), null)) {
+                    internalSchedule(timerJobInstance);
```

Comment:
Could it throw an exception that in the previous implementation was caught?
`logger.warn("Could not schedule the interval trigger {}", ejbTimerJob.getTimerJobInstance(), e1);`